docs/source/upgrade_to_newest_version.md

```diff
@@ -0,0 +1,164 @@
+
+If you did not vendor the shim in your v1.4 chaincode, the old v1.4 chaincode
+images will still technically work after upgrade, but you are in a risky state.
+If the chaincode image gets deleted from your environment for whatever reason,
+the next invoke on v2.0 peer will try to rebuild the chaincode image and you'll
+get an error that the shim cannot be found.
+
+At this point, you have two options:
+
+1. If the entire channel is ready to upgrade chaincode, you can upgrade the
+   chaincode on all peers and on the channel (using either the old or new
+   lifecycle depending on the `Application` capability level you have enabled).
+   The best practice at this point would be to vendor the new Go chaincode shim
+   using modules.
+
+2. If the entire channel is not yet ready to upgrade the chaincode, you can use
+   peer environment variables to specify the v1.4 chaincode environment `ccenv`
+   be used to rebuild the chaincode images. This v1.4 `ccenv` should still work
+   with a v2.0 peer.
+
+## Chaincode logger (Go chaincode only)
+
+Support for user chaincodes to utilize the chaincode shim's logger via `NewLogger()`.
+Chaincodes that used the shim's `NewLogger()` must now shift to their own
+preferred logging mechanism.
+
+For more information, check out [Logging control](./logging-control.html#Chaincode).
+
+## Chaincode lifecycle
+
+For more information about the new chaincode lifecycle, check out [Chaincode for operators](./chaincode4noah.html).
+
+It is a best practice to upgrade all of the peers on a channel before enabling
+the `Channel` and `Channel/Application` capabilities that enable the new chaincode
+lifecycle to function. Note that any peers that are not at v2.0 will crash after
+enabling either capability, while any ordering nodes that are not at v2.0 will
+crash after the `Channel` capability has been enabled. This crashing behavior is
+intentional as it prevents the node from rejecting the config block (given that
+it contains a value it does not understanding) when nodes at the appropriate
+node level would accept it.
+
+After the `Application` capability has been updated to `V2_0`, you must use
+the v2.0 lifecycle procedures to package, install, approve, and commit new
+chaincodes on the channel. As a result, make sure to be prepared for the new
+lifecycle before updating the capability.
+
+The new lifecycle defaults to using the endorsement policy configured in the
+channel config (e.g., a `MAJORITY` of orgs). Therefore this endorsement policy
+should be added to the channel configuration when enabling capabilities on the
+channel.
+
+For information about how to edit the relevant channel configurations to enable
+the new lifecycle, check out [Enabling the new chaincode lifecycle](./enable_cc_lifecycle.html).
+
+It is also a best practice to add an endorsement policy for each organization,
+which will also define how changes will be made at the organization level.
```

Comment:
Suggest switching the order of the above 2 paragraphs. In addition, it is `required` (not only a best practice) to add an endorsement policy for each organization.

Reply:
Have fixed this to reflect that it's required.